### ISSUE TITLE
SSD_inception_V2 paper architecture

### ISSUE THREAD

**Issue by 89douner** [tensorflow/models#5426]
System information
What is the top-level directory of the model you are using: /home/user
Have I written custom code: No
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
TensorFlow installed from (source or binary): source
TensorFlow version (use command below): 1.10.1
Bazel version (if compiling from source): I don't use it
CUDA/cuDNN version: CUDA 9.0 / cuDNN: 7.1
GPU model and memory: Geforce GTX 1070
Exact command to reproduce: No

[Question]
Paper: Speed/accuracy trade-offs for modern convolutional object detectors

The paper says that "We use Mixed 4c and Mixed 5c, appending four additional convolutional layers with
decaying resolution with depths 512, 256, 256, 128 respectively." But, I didn't understand the architecture exactly. This means the order of Mixed 4c -> Mixed 5c -> 512 -> 256 -> 256 -> 128 ?? I also referred code, but still didn't. Just, I know that SSD_inception_V2 feature extractor structure ('Conv2d_1a_7x7', 'MaxPool_2a_3x3', 'Conv2d_2b_1x1', 'Conv2d_2c_3x3', 'MaxPool_3a_3x3', 'Mixed_3b', 'Mixed_3c', 'Mixed_4a', 'Mixed_4b', 'Mixed_4c', 'Mixed_4d', 'Mixed_4e', 'Mixed_5a', 'Mixed_5b', 'Mixed_5c').

https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v2.py

[Reference code site]
Keyword: from_layer

https://github.com/tensorflow/models/blob/1f484095c0981e2a62403b16256cb877749dfe94/research/object_detection/models/feature_map_generators.py
https://github.com/tensorflow/models/blob/master/research/object_detection/models/ssd_inception_v2_feature_extractor.py

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by netanel-s** [comment]
InceptionV2 is only the backbone, and it has the layers Mixed 4c and Mixed 5c.
SSD doesn't only take layers of the backbone, but also "SSD layers", which come after it with decaying resolution, which are used to detect larger and larger objects.
This can be visualized in Figure 2 (top) in the SSD paper: https://arxiv.org/pdf/1512.02325.pdf 
So, yes, the layers used for detection are Mixed 4c, Mixed 5c, and then 4 layers with decaying resolution, and with amount of channels 512, 256, 256, 128 respectively.

**Comment by 89douner** [comment]
I already knew what you talk about. As I mentioned, CNN backbone of SSD_inception_V2 has the order of 'Conv2d_1a_7x7', 'MaxPool_2a_3x3', 'Conv2d_2b_1x1', 'Conv2d_2c_3x3', 'MaxPool_3a_3x3', 'Mixed_3b', 'Mixed_3c', 'Mixed_4a', 'Mixed_4b', 'Mixed_4c', 'Mixed_4d', 'Mixed_4e', 'Mixed_5a', 'Mixed_5b', 'Mixed_5c'.

If you go to this [link](https://github.com/tensorflow/models/blob/master/research/object_detection/models/ssd_inception_v2_feature_extractor.py), you can see below code.
`feature_map_layout = {
        'from_layer': ['Mixed_4c', 'Mixed_5c', '', '', '', ''],
        'layer_depth': [-1, -1, 512, 256, 256, 128],
        'use_explicit_padding': self._use_explicit_padding,
        'use_depthwise': self._use_depthwise,
    }`


if you go to [this link](https://github.com/tensorflow/models/blob/1f484095c0981e2a62403b16256cb877749dfe94/research/object_detection/models/feature_map_generators.py), you can see the algorithm. 

`for index, from_layer in enumerate(feature_map_layout['from_layer']):
      net = []
      self.convolutions.append(net)
      layer_depth = feature_map_layout['layer_depth'][index]
      conv_kernel_size = 3
      if 'conv_kernel_size' in feature_map_layout:
        conv_kernel_size = feature_map_layout['conv_kernel_size'][index]
      if from_layer:
        base_from_layer = from_layer
      else:
        if insert_1x1_conv:
          layer_name = '{}_1_Conv2d_{}_1x1_{}'.format(
              base_from_layer, index, depth_fn(layer_depth / 2))
          net.append(tf.keras.layers.Conv2D(depth_fn(layer_depth / 2),
                                            [1, 1],
                                            padding='SAME',
                                            strides=1,
                                            name=layer_name + '_conv',
                                            **conv_hyperparams.params()))
          net.append(
              conv_hyperparams.build_batch_norm(
                  training=(is_training and not freeze_batchnorm),
                  name=layer_name + '_batchnorm'))
          net.append(
              conv_hyperparams.build_activation_layer(
                  name=layer_name))`

After seeing the code, I still don't understand that "We use Mixed 4c and Mixed 5c, appending four additional convolutional layers with decaying resolution with depths 512, 256, 256, 128 respectively". 
(I just wander the location of Mixed 4c and Mixed 5c.)

![kakaotalk_20181003_165308857](https://user-images.githubusercontent.com/31752297/46397335-f18f7800-c72c-11e8-9f00-2318f26ef948.jpg)

**Comment by 89douner** [comment]
Thanks for your answer!

**Comment by karmel** [comment]
Has this issue been resolved?

**Comment by 89douner** [comment]
If you have another answer, you can write! please! 
@karmel

**Comment by karmel** [comment]
If the answer here is insufficient, I recommend you ask on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since this is not a bug or feature request. There is a larger community that reads questions there, but my expertise is not the particular model architectures here.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
